src/output/util/format_type.js

```diff
@@ -139,21 +140,21 @@ export default function formatType(getHref, node) {
 
     case Syntax.FieldType:
       if (node.value) {
-        return [t(node.key + ': ')].concat(formatType(getHref, node.value));
+        return [t(`${node.key}: `)].concat(formatType(getHref, node.value));
       }
       return [t(node.key)];
 
     case Syntax.FunctionType:
       result = [t('function (')];
 
-      if (node['this']) {
-        if (node['new']) {
+      if (node.this) {
```

Comment:
I would keep it they way they had it before, for clarity. Reserved words should be handled with care